runtime/src/consensus/tendermint/verifier/store/lru.rs

```diff
@@ -114,40 +115,44 @@ impl LightStore for LruStore {
 
     fn highest(&self, status: Status) -> Option<LightBlock> {
         self.inner()
-            .store(status)
             .blocks
             .iter()
+            .filter(|(_, e)| e.status == status)
             .max_by_key(|(&height, _)| height)
-            .map(|(_, lb)| lb.clone())
+            .map(|(_, e)| e.light_block.clone())
     }
 
     fn highest_before(&self, height: Height, status: Status) -> Option<LightBlock> {
         self.inner()
-            .store(status)
             .blocks
             .iter()
+            .filter(|(_, e)| e.status == status)
             .filter(|(h, _)| h <= &&height)
             .max_by_key(|(&height, _)| height)
-            .map(|(_, lb)| lb.clone())
+            .map(|(_, e)| e.light_block.clone())
     }
 
     fn lowest(&self, status: Status) -> Option<LightBlock> {
-        let mut inner = self.inner();
-        let store = inner.store(status);
+        let mut store = self.inner();
 
-        store
-            .lowest_height
-            .and_then(|lowest_height| store.blocks.get(&lowest_height).cloned())
+        store.lowest_height.and_then(|lowest_height| {
+            store
+                .blocks
+                .get(&lowest_height)
```

Comment:
This will return lowest block for trusted status only (or whatever is the status of the lowest block). For others, this will return none.
I would revert this to previous version (copy-paste-edit from `highest`).